### ISSUE TITLE
Appending a class to an H1-tag breaks the ToC-hack

### ISSUE THREAD

**Issue by martinbira** [squidfunk/mkdocs-material#2563]
- [X] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [X] ... the problem doesn't occur with the default MkDocs template
- [X] ... the problem is not in any of my customizations (CSS, JS, template)
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

## Description

When appeding a class to an H1 tag with attribute lists, the ToC-hack in the `toc.html` partial breaks and the first item in the ToC is the page title. The ToC-hack assumes that the first attribute in an H1 is `id`, but when adding a class to it, `class` is the first tag.

### Expected behavior

When adding a class to an H1 using attribute lists the Page Title should not be in the ToC.

### Actual behavior

When adding a class to an H1 using attribute lists the Page Title is the first ToC item.

### Steps to reproduce the bug

1. Ensure that the `attr_list` markdown extension is used.
2. Add a class to the H1 in a `.md` file with at least one H2:
```
# This is my title { .myTitleClass }

Text

## This is my first ToC entry

Text
```
3. Run `mkdocs serve` and view 

### Package versions

* Python: 3.9.0
* MkDocs: 1.1.2
* Material: 7.1.1

### Project configuration

``` yaml
markdown_extensions:
  - attr_list
```

### System information

N/A

### Temporary workaround

A temporary workaround is to add an additional `if` to the `toc.html` partial that checks for `class`:
```
  {% if toc | first is defined and "\x3ch1 class=" in page.content %}
    {% set toc = (toc | first).children %}
  {% endif %}
```

  [1]: https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md

**Comment by squidfunk** [comment]
I think I have a fix in 8d752312a. It was a great opportunity to revisit the hack I introduced in 2016. I found out that we can use the `level` property of the table of contents item, which tells us if the first entry is a `h1`.

**Comment by martinbira** [comment]
That was a pretty sweet solution, and using the `iterable`. Nice job!

**Comment by squidfunk** [comment]
Released as part of 7.1.2.
